mne/realtime/tests/test_mockclient.py

```diff
@@ -180,8 +210,123 @@ def test_find_events():
+
+    rt_epochs.start()
+    rt_client.send_data(rt_epochs, picks, tmin=0, tmax=raw_tmax,
+                        buffer_size=buffer_size)
+
+    assert len(rt_epochs) == len(epochs_to_keep)
+    assert_array_equal(rt_epochs.drop_log, expected_drop_log)
+    assert_array_equal(rt_epochs.selection, epochs_to_keep)
+    rt_data = rt_epochs.get_data()
+    assert rt_data.shape == epochs_data.shape
+    assert_array_equal(rt_data, epochs_data)
+
+
+@sample.requires_sample_data
+def test_events_sampledata():
+    """ based on examples/realtime/plot_compute_rt_decoder.py"""
+    data_path = sample.data_path()
+    raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+    raw = mne.io.read_raw_fif(raw_fname, preload=True)
+    raw_tmin, raw_tmax = 0, 90
+
+    tmin, tmax = -0.2, 0.5
+    event_id = dict(aud_l=1, vis_l=3)
+
+    # select gradiometers
+    picks = mne.pick_types(raw.info, meg='grad', eeg=False, eog=True,
+                           stim=True, exclude=raw.info['bads'])
+
+    # load data with usual Epochs for later verification
+    raw_cropped = raw.copy().crop(raw_tmin, raw_tmax)
+    events_offline = mne.find_events(raw_cropped)
+    epochs_offline = Epochs(raw_cropped, events_offline, event_id=event_id,
+                            tmin=tmin, tmax=tmax, picks=picks, decim=1,
+                            reject=dict(grad=4000e-13, eog=150e-6),
+                            baseline=None)
+    epochs_offline.drop_bad()
+
+    # create the mock-client object
+    rt_client = MockRtClient(raw)
+    rt_epochs = RtEpochs(rt_client, event_id, tmin, tmax, picks=picks, decim=1,
+                         reject=dict(grad=4000e-13, eog=150e-6), baseline=None,
+                         isi_max=1.)
+
+    rt_epochs.start()
+    rt_client.send_data(rt_epochs, picks, tmin=raw_tmin, tmax=raw_tmax,
+                        buffer_size=1000)
+
+    expected_events = epochs_offline.events.copy()
+    expected_events[:, 0] = expected_events[:, 0] - raw_cropped.first_samp
+    assert np.all(expected_events[:, 0] <=
+                  (raw_tmax - tmax) * raw.info['sfreq'])
+    assert_array_equal(rt_epochs.events, expected_events)
+    assert len(rt_epochs) == len(epochs_offline)
+
+    data_picks = mne.pick_types(epochs_offline.info, meg='grad', eeg=False,
+                                eog=True,
+                                stim=False, exclude=raw.info['bads'])
+
+    for ev_num, ev in enumerate(rt_epochs.iter_evoked()):
+        if ev_num == 0:
+            X_rt = ev.data[None, data_picks, :]
+            y_rt = int(ev.comment)  # comment attribute contains the event_id
+        else:
+            X_rt = np.concatenate((X_rt, ev.data[None, data_picks, :]), axis=0)
+            y_rt = np.append(y_rt, int(ev.comment))
+
+    X_offline = epochs_offline.get_data()[:, data_picks, :]
+    y_offline = epochs_offline.events[:, 2]
+    assert_array_equal(X_rt, X_offline)
+    assert_array_equal(y_rt, y_offline)
```

Comment:
I think this is a great idea! We should probably do this also for `FieldTripClient`...